hydra-node/src/Hydra/Network/Ouroboros.hs

```diff
@@ -142,20 +145,35 @@ withOuroborosNetwork tracer localHost remoteHosts networkCallback between = do
       (info : _) -> pure $ addrAddress info
       _ -> error "getAdrrInfo failed.. do proper error handling"
 
-  connect iomgr newBroadcastChannel app = do
+  -- connect will automatically restart either
+  -- if the lost of peers is modified or if the subscription worker dies.
```

Comment:
We could keep the same semantics by not using race_, but withAsync on the subscription thread and link (https://open.spotify.com/show/0cPsvdqTreF6sKg6VwSrMl?si=T-j6w7PFTTuAQ31TMQns_A&utm_source=copy-link)

Reply:
👍 🎵

Reply:
As per the docs, `link` role is to ensure exceptions from the thread are delivered to whatever other thread it gets linked with, but does not terminate the inner action when the thread terminates normally (or gets cancelled for that matter). Seems like `race_` is really what we want here.

Reply:
We could add a `wait` but then this makes things more complicated, and the `race_` semantics is just what we need here AFAIU